modules/linear_boltzmann_solvers/lbs_solver/lbs_solver.cc

```diff
@@ -646,23 +628,33 @@ LBSSolver::SetOptions(const InputParameters& params)
     else if (spec.Name() == "read_restart_data")
       options_.read_restart_data = spec.GetValue<bool>();
 
-    else if (spec.Name() == "read_restart_folder_name")
-      options_.read_restart_folder_name = spec.GetValue<std::string>();
-
-    else if (spec.Name() == "read_restart_file_base")
-      options_.read_restart_file_base = spec.GetValue<std::string>();
+    else if (spec.Name() == "read_restart_directory_name")
+      options_.read_restart_directory_name = spec.GetValue<std::string>();
 
-    else if (spec.Name() == "write_restart_data")
-      options_.write_restart_data = spec.GetValue<bool>();
+    else if (spec.Name() == "read_restart_file_stem")
+      options_.read_restart_file_stem = spec.GetValue<std::string>();
 
-    else if (spec.Name() == "write_restart_folder_name")
-      options_.write_restart_folder_name = spec.GetValue<std::string>();
+    else if (spec.Name() == "write_restart_directory_name")
+      options_.write_restart_directory_name = spec.GetValue<std::string>();
 
-    else if (spec.Name() == "write_restart_file_base")
-      options_.write_restart_file_base = spec.GetValue<std::string>();
+    else if (spec.Name() == "write_restart_file_stem")
+      options_.write_restart_file_stem = spec.GetValue<std::string>();
 
-    else if (spec.Name() == "write_restart_interval")
-      options_.write_restart_interval = spec.GetValue<double>();
+    else if (spec.Name() == "write_restart_time_interval")
+    {
+      options_.write_restart_time_interval = spec.GetValue<int>();
+      if (not std::filesystem::is_directory(options_.write_restart_directory_name))
+      {
+        if (opensn::mpi_comm.rank() == 0)
+          std::filesystem::create_directory(options_.write_restart_directory_name);
+        opensn::mpi_comm.barrier();
+        if (not std::filesystem::is_directory(options_.write_restart_directory_name))
+        {
+          throw std::runtime_error("Failed to create restart directory " +
+                                   options_.write_restart_directory_name);
+        }
+      }
+    }
```

Comment:
This seems to be more appropriate with `write_restart_directory_name`, FYI.

Reply:
Fixed.